### DISCUSSION TITLE
How can I set "gitlab_credentials" using a docker-compose instance?

### DISCUSSION THREAD

**Discussion by calbasi** [WeblateOrg/weblate#8128] [Q&A]
Hi!

I've tried these on my docker-compose.override.yml file:

      WEBLATE_GITLAB_CREDENTIALS: git.duniter.org: username: XXXXXXXX,token: XXXXXXXXXXXXXXX

I get a new error:

ERROR: yaml.scanner.ScannerError: mapping values are not allowed here
  in "./docker-compose.override.yml", line 17, column 50

So I wonder how can I set gitlab_credentials setting. Maybe using the GUI? (/admin/configuration/setting/add/ ). If so, which format should I use?

**Comment by nijel** [comment]
The error you get is from the docker-compose YAML parser, to have a valid YAML syntax it should be:

```yaml
  WEBLATE_GITLAB_CREDENTIALS: "git.duniter.org: username: XXXXXXXX,token: XXXXXXXXXXXXXXX"
```

Still, this setting is currently not parsed from the environment, see https://docs.weblate.org/en/latest/admin/install/docker.html#docker-environment-variables for list of supported variables (you can use WEBLATE_GITLAB_USERNAME and WEBLATE_GITLAB_TOKEN).

**Comment by nijel** [comment]
Also posted as https://github.com/WeblateOrg/weblate/issues/8132
